prince_cr/solvers/propagation.py

```diff
@@ -113,13 +113,15 @@ def get_solution_group(self, nco_ids, epow=3, egrid=None):
         spectra = np.zeros((len(nco_ids), com_egrid.size))
         for idx, pid in enumerate(nco_ids):
             curr_egrid, curr_spec = self.get_solution_scale(pid, epow=epow)
-            res = np.exp(
-                np.interp(
-                    np.log(com_egrid),
-                    np.log(curr_egrid),
-                    np.log(curr_spec),
-                    left=np.nan,
-                    right=np.nan))
+            mask = curr_spec > 0.
+            if np.count_nonzero(mask) > 0:
+                res = np.exp(
+                    np.interp(
+                        np.log(com_egrid),
+                        np.log(curr_egrid[mask]),
+                        np.log(curr_spec[mask]),
+                        left=np.nan,
+                        right=np.nan))
             spectra[idx] = np.nan_to_num(res)
```

Comment:
This change is to fix the numpy overflow warnings, right? But what happens if curr_spec is all `0`? Then the if statement it skipped and l.125 will throw an error. 

I think this should look like this (have not tested):
```suggestion
            mask = curr_spec > 0.
            if np.count_nonzero(mask) > 0:
                res = np.exp(
                    np.interp(
                        np.log(com_egrid),
                        np.log(curr_egrid[mask]),
                        np.log(curr_spec[mask]),
                        left=np.nan,
                        right=np.nan))
            else:
                res = np.zeros_like(com_egrid)
            spectra[idx] = np.nan_to_num(res)
```